src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,155 @@
+    super(client, data);
+
+    this.syncHandle = syncHandle;
+
+    /**
+     * The ID of the invoked command.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.commandID = data.data.id;
+
+    /**
+     * The name of the invoked command.
+     * @type {string}
+     * @readonly
+     */
+    this.commandName = data.data.name;
+
+    const map = o => {
+      switch (o.type) {
+        case ApplicationCommandOptionType.SUB_COMMAND:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            options: o.options.map(map),
+          };
+        case ApplicationCommandOptionType.SUB_COMMAND_GROUP:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            options: o.options.map(map),
+          };
+        case ApplicationCommandOptionType.STRING:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.INTEGER:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.BOOLEAN:
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            value: o.value,
+          };
+        case ApplicationCommandOptionType.USER: {
+          const rawUser = data.data.resolved.users[o.value];
+          const rawMember = data.data.resolved.members[o.value];
+          if (rawMember) {
+            rawMember.user = rawUser;
+          }
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            user: this.client.users ? this.client.users.add(rawUser, false) : new User(this.client, rawUser),
+            member: rawMember ? this.guild?.members.add(rawMember, false) ?? null : null,
+          };
+        }
+        case ApplicationCommandOptionType.CHANNEL: {
+          const raw = data.data.resolved.channels[o.value];
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            channel: this.client.channels
+              ? this.client.channels.add(raw, false)
+              : Channel.create(this.client, raw, this.guild),
+          };
+        }
+        case ApplicationCommandOptionType.ROLE: {
+          const raw = data.data.resolved.roles[o.value];
+          return {
+            name: o.name,
+            type: ApplicationCommandOptionType[o.type],
+            role: this.guild ? this.guild.roles.add(raw, false) : new Role(this.client, raw, undefined),
+          };
+        }
+        default:
+          return o;
+      }
+    };
+    /**
+     * The options passed to the command.
+     * @type {Object}
+     * @readonly
+     */
+    this.options = data.data.options.map(map);
```

Comment:
```suggestion
    this.options = data.data.options?.map(map);
```
`options` can be undefined when a command doesn't have parameters (or none are given when executing the command)

Reply:
```suggestion
    this.options = data.data.options?.map(map) ?? [];
```

Wouldn't it be better if `options` was an empty array, instead of just being undefined?